shopinvader/models/shopinvader_variant.py

```diff
@@ -50,20 +50,26 @@ class ShopinvaderVariant(models.Model):
 
     @contextmanager
     @api.multi
-    def _action_product_disabled(self):
+    def _action_product_disabled_enabled(self):
         """
         Action a deactivation of a variant, if every variants are disabled:
         disable the product too.
+        Also when a variant is enabled, the related shopinvader product
+        should be re-enabled too.
         :return:
         """
         product_active_dict = {
             p: p.active for p in self.mapped("shopinvader_product_id")
         }
         yield
         for variant in self:
+            shopinv_product = variant.shopinvader_product_id
             if variant.active:
+                # If the variant is active and the related shop. product is
+                # not active, we have to active it.
+                if not shopinv_product.active:
+                    shopinv_product.write({"active": True})
```

Comment:
can't we collect all products that need an update and call `write` only once on all of them?